pkgs/tools/networking/openssh/default.nix

```diff
@@ -18,124 +19,148 @@
+
+    postPatch =
+      # On Hydra this makes installation fail (sometimes?),
+      # and nix store doesn't allow such fancy permission bits anyway.
+      ''
+        substituteInPlace Makefile.in --replace '$(INSTALL) -m 4711' '$(INSTALL) -m 0711'
+      '';
+
+    nativeBuildInputs = [ pkg-config ]
+      ++ optional (hpnSupport || withGssapiPatches) autoreconfHook
+      ++ optional withKerberos pkgs.kerberos.dev;
+    buildInputs = [ zlib openssl libedit pam ]
+      ++ optional withFIDO libfido2
+      ++ optional withKerberos kerberos;
+
+    preConfigure = ''
+      # Setting LD causes `configure' and `make' to disagree about which linker
+      # to use: `configure' wants `gcc', but `make' wants `ld'.
+      unset LD
+    ''
+    # Upstream build system does not support static build, so we fall back
+    # on fragile patching of configure script.
+    #
+    # libedit is found by pkg-config, but without --static flag, required
+    # to get also transitive dependencies for static linkage, hence sed
+    # expression.
+    #
+    # Kerberos can be found either by krb5-config or by fall-back shell
+    # code in openssh's configure.ac. Neither of them support static
+    # build, but patching code for krb5-config is simpler, so to get it
+    # into PATH, kerberos.dev is added into buildInputs.
+    + optionalString stdenv.hostPlatform.isStatic ''
+      sed -i "s,PKGCONFIG --libs,PKGCONFIG --libs --static,g" configure
+      sed -i 's#KRB5CONF --libs`#KRB5CONF --libs` -lkrb5support -lkeyutils#g' configure
+      sed -i 's#KRB5CONF --libs gssapi`#KRB5CONF --libs gssapi` -lkrb5support -lkeyutils#g' configure
+    '';
 
+    # I set --disable-strip because later we strip anyway. And it fails to strip
+    # properly when cross building.
+    configureFlags = [
+      "--sbindir=\${out}/bin"
+      "--localstatedir=/var"
+      "--with-pid-dir=/run"
+      "--with-mantype=man"
+      "--with-libedit=yes"
+      "--disable-strip"
+      (if pam != null then "--with-pam" else "--without-pam")
+    ] ++ optional (etcDir != null) "--sysconfdir=${etcDir}"
+      ++ optional withFIDO "--with-security-key-builtin=yes"
+      ++ optional withKerberos (assert kerberos != null; "--with-kerberos5=${kerberos}")
+      ++ optional stdenv.isDarwin "--disable-libutil"
+      ++ optional (!linkOpenssl) "--without-openssl";
+
+    buildFlags = [ "SSH_KEYSIGN=ssh-keysign" ];
+
+    enableParallelBuilding = true;
+
+    hardeningEnable = [ "pie" ];
+
+    postInstall = ''
+      ln -s ${ssh-copy-id}/bin/ssh-copy-id $out/bin/ssh-copy-id
+    '';
+
+    installTargets = [ "install-nokeys" ];
+    installFlags = [
+      "sysconfdir=\${out}/etc/ssh"
+    ];
+
+    meta = withDescription "An implementation of the SSH protocol";
+  };
+
+  ssh-copy-id = stdenv.mkDerivation {
+    pname = "ssh-copy-id";
+    inherit src version;
+
+    phases = [ "unpackPhase" "patchPhase" "installPhase" "fixupPhase" ];
```

Comment:
```suggestion
```

Please disable phases which should not run with ``dont... = true;``

Reply:
Fixed.